tools/m2-repo-cleaner/src/main/java/org/fusesource/ide/tools/mavenutils/M2RepoChecker.java

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.fusesource.ide.tools.mavenutils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+/**
+ * @author lheinema
+ */
+public class M2RepoChecker {
+
+	public static final String DEFAULT_REPO_FOLDER = System.getProperty("user.home") + File.separator + ".m2" + File.separator + "repository" + File.separator;
+	
+	private static long checkedFiles = 0;
+	private static long checkedFolders = 0;
+	private static List<String> corruptedFiles = new ArrayList<>();
+	
+	/**
+	 * @param args
+	 */
+	public static void main(String[] args) {
+		String repoPath = DEFAULT_REPO_FOLDER;
+		if (args.length>0) {
+			repoPath = args[0];
+			System.out.println("Using custom repo folder: " + repoPath);
+		} else {
+			System.out.println("Using default repo folder: " + repoPath);
+		}
+		
+		File repoFolder = new File(repoPath);
+		if (!repoFolder.exists() || !repoFolder.isDirectory()) {
+			System.err.println("The given folder does not exist or is not a directory. Also make sure you have the rights to access it.");
+			System.exit(0);
```

Comment:
calling exit(0) can be dangerous.
Are you sure that the Java program is launched in a forked process? otherwise ti will kill the maven instance which has launched it.